src/beman/dump/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+add_library(beman.dump STATIC)
+add_library(beman::dump ALIAS beman.dump)
+
+target_sources(beman.dump PRIVATE dump.cpp)
+
+target_include_directories(
+  beman.dump
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../../include>)
+target_compile_features(
+  beman.dump
+  PUBLIC cxx_std_23)
+
+install(
+  TARGETS beman.dump
+  EXPORT beman.dump
+  DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../../../include/
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  FILES_MATCHING
+  PATTERN "*.hpp")
+
+if(BUILD_TESTING)
+  include(GoogleTest)
+
+  add_executable(beman.dump.tests)
+  target_sources(beman.dump.tests PRIVATE dump.t.cpp)
+  target_link_libraries(beman.dump.tests
+                        PRIVATE beman::dump GTest::gtest GTest::gtest_main)
+
+  gtest_add_tests(beman.dump.tests "" AUTO)
```

Comment:
I think we need some kind of test that verifies that the `dump()` call is implemented correctly. One way this could be done is to:

1. Write a short CMake script that runs a program that calls `dump()`, pipes it to a file, and verifies its output
2. Register the CMake execution as a test using `add_test`

Reply:
Defered to #2

Reply:
It is almost trivial to redirect std::cout though, I don't see the need to write to a file, but we can discuss this later.